src/test/java/org/candlepin/subscriptions/security/auth/AdminOnlyTest.java

```diff
@@ -0,0 +1,144 @@
+import org.springframework.security.access.AccessDeniedException;
+import org.springframework.security.config.annotation.method.configuration.EnableGlobalMethodSecurity;
+import org.springframework.security.core.Authentication;
+import org.springframework.test.context.junit.jupiter.SpringJUnitConfig;
+
+@SpringJUnitConfig(classes = AdminOnlyConfiguration.class)
+public class AdminOnlyTest {
+
+    @Autowired
+    ApplicationContext context;
+
+    @EnableGlobalMethodSecurity(prePostEnabled = true)
+    protected static class AdminOnlyConfiguration {
+        @Bean
+        public StubResource stubResource() {
+            return new StubResource();
+        }
+
+        @Bean
+        public ApplicationProperties applicationProperties() {
+            return new ApplicationProperties();
+        }
+
+        @Bean
+        public WhitelistedAccountReportAccessService reportAccessService() {
+            return mock(WhitelistedAccountReportAccessService.class);
+        }
+    }
+
+    protected static class StubResource {
+        @AdminOnly
+        public void adminOnlyCall() {
+            // Does nothing
+        }
+
+        @ReportingAdminOnly
+        public void reportingAdminOnlyCall() {
+            // Does nothing
+        }
+    }
+
+    @Test
+    @WithMockRedHatPrincipal(value = "123456", roles = {})
+    void testAdminOnlyCallWithNonAdmin() throws Exception {
+        StubResource stub = context.getBean(StubResource.class);
+
+        context.getBean(ApplicationProperties.class).setOrgAdminRequired(true);
+        assertThrows(AccessDeniedException.class, stub::adminOnlyCall);
+
+        context.getBean(ApplicationProperties.class).setOrgAdminRequired(false);
+        assertDoesNotThrow(stub::adminOnlyCall);
+    }
+
+    @Test
+    @WithMockRedHatPrincipal("123456")
+    void testAdminOnlyCallWithOrgAdmin() throws Exception {
+        StubResource stub = context.getBean(StubResource.class);
+
+        context.getBean(ApplicationProperties.class).setOrgAdminRequired(true);
+        assertDoesNotThrow(stub::adminOnlyCall);
+
+        context.getBean(ApplicationProperties.class).setOrgAdminRequired(false);
+        assertDoesNotThrow(stub::adminOnlyCall);
+    }
+
+    @Test
+    @WithMockRedHatPrincipal(value = "123456", roles = {})
+    void testReportingAdminOnlyCallWithNonAdmin() throws Exception {
+        StubResource stub = context.getBean(StubResource.class);
+        WhitelistedAccountReportAccessService mockAccess =
+            context.getBean(WhitelistedAccountReportAccessService.class);
+        when(mockAccess.providesAccessTo(any(Authentication.class))).thenReturn(false);
+
+        context.getBean(ApplicationProperties.class).setOrgAdminRequired(true);
+        assertThrows(AccessDeniedException.class, stub::reportingAdminOnlyCall);
+
+        context.getBean(ApplicationProperties.class).setOrgAdminRequired(false);
+        assertDoesNotThrow(stub::reportingAdminOnlyCall);
```

Comment:
This should still throw, as the reportingAccessService shows they're not opted-in yet.